x/ccv/consumer/keeper/keeper.go

```diff
@@ -593,48 +593,80 @@ func (k Keeper) GetAllValidators(ctx sdk.Context) (validators []stakingtypes.Val
 	return validators
 }
 
-// SetPendingPackets sets the pending CCV packets
-func (k Keeper) SetPendingPackets(ctx sdk.Context, packets ccv.ConsumerPacketDataList) {
+// getAndIncrementPendingPacketsIdx returns the current pending packets index and increments it.
+// This index is used for implementing a FIFO queue of pending packets in the KV store.
+func (k Keeper) getAndIncrementPendingPacketsIdx(ctx sdk.Context) (toReturn uint64) {
 	store := ctx.KVStore(k.storeKey)
-	bz, err := packets.Marshal()
-	if err != nil {
-		// This should never happen
-		panic(fmt.Errorf("failed to marshal ConsumerPacketDataList: %w", err))
+	bz := store.Get(types.PendingPacketsIndexKey())
+	if bz == nil {
+		toReturn = 0
+	} else {
+		toReturn = sdk.BigEndianToUint64(bz)
 	}
```

Comment:
This condition can be simplified as `toReturn` is already equal to 0.
```suggestion
	if bz == nil {
		toReturn = 0
	} else {
		toReturn = sdk.BigEndianToUint64(bz)
	}
```